relay-event-schema/src/protocol/span.rs

```diff
@@ -97,6 +97,10 @@ pub struct Span {
     #[metastructure(skip_serialization = "empty")]
     pub platform: Annotated<String>,
 
+    /// Whether the span is a segment span that was converted from a transaction.
+    #[metastructure(skip_serialization = "empty")]
+    pub was_transaction: Annotated<bool>,
```

Comment:
This effectively adds this field to the Span schema. Relays will accept this from other Relays (and SDKs) as well as forward it.

Is this the best way to deal with the problem and something we are okay with?

Reply:
I think so, yes. I want an explicit flag to represent "this is a span extracted from a transaction", which gets serialized into the payload so the next Relay in the chain can check it as well. We will also need this for https://github.com/getsentry/relay/issues/3278. If an untrusted source sets this flag, it will only mess up their own data.

As for the schema, we already accept any field in the span payload, and the schema is not automatically documented, so adding a field in Relay does not really make a difference:
https://github.com/getsentry/relay/blob/f820884cdc4af39ecd49c3bc5fdf1bc436184283/relay-event-schema/src/protocol/span.rs#L104-L107
https://github.com/getsentry/relay/blob/f820884cdc4af39ecd49c3bc5fdf1bc436184283/relay-event-schema/src/protocol/event.rs#L467-L468